### PR TITLE
Dexscreener Emojicoin Support

### PR DESCRIPTION
**Pull request by CapCap**
Adds `block_number` and the `event_index` to liquidity and swap events. 

This can help make it easier to debug, and models the ‘correct’ way to represent blocks/transactions/events on chain to external parties like dexscreener. I was initially trying to avoid this by being a bit hacky, but I do worry about setting weird precedents when we start with dexscreener could end up with weird behavior when other projects eventually integrate too.

This is explicitly not backwards compatible.

### CODE REVIEW

**Review by CRBl69 — CHANGES_REQUESTED** [review]
This looks mainly good to me. The only thing is that I do not like the fact that `block_number` and `evnet_index` are options/nullable. I would rather set them as `NOT NULL` and remove the option logic. This would make the processor impossible to upgrade, and we'd have to deploy a new stack. I think this is ok, as the backfilling won't take too much, and we can change the backend that the frontend points to very easily. @alnoki @xbtmatt opinions on this ?

**Comment by alnoki** [comment]
> I would rather set them as NOT NULL and remove the option logic

@CapCap I agree with @CRBl69 's assessment on this

Backwards compatibility is not necessarily a requirement this early, so is cleanest to simply start a new deployment with breaking logic then have any calling resources point at the new stack once it has backfilled

**Comment by CapCap** [comment]
@alnoki @CRBl69 if that's the prefered way of doing it, works for me, less pain on my end :-) 

@xbtmatt I guess I won't be doing the backfill BQ stuff then- will just make it mandatory from start

File: rust/processor/src/db/postgres/migrations/2024-12-03-050428_add_block_number_and_event_index/up.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- Your SQL goes here
+ALTER TABLE swap_events
+        ADD COLUMN block_number BIGINT NOT NULL,
+        ADD COLUMN event_index NUMERIC NOT NULL;
+CREATE INDEX se_block_num ON swap_events (block_number);
+
+ALTER TABLE liquidity_events
+        ADD COLUMN block_number BIGINT NOT NULL,
+        ADD COLUMN event_index NUMERIC NOT NULL,
```
Comment (CRBl69) [inline rust/processor/src/db/postgres/migrations/2024-12-03-050428_add_block_number_and_event_index/up.sql]:
```suggestion
        ADD COLUMN event_index NUMERIC NOT NULL;
```

File: rust/processor/src/db/postgres/schema.rs [hunk 2]
```diff
@@ -957,7 +957,10 @@ diesel::table! {
     use super::sql_types::TriggerType;
 
     liquidity_events (market_id, market_nonce) {
+        block_number -> Int8,
         transaction_version -> Int8,
+        event_index -> Int8,
+
```
Comment (CRBl69) [inline rust/processor/src/db/postgres/schema.rs]: It seems you have manually added this lines.

This file is actually generated by the `diesel` CLI whenever you run migrations for the first time.

I suggest you start a clean DB and run the migrations to update this file (right now, the int types are incorrect).

You can use these commands:

```sh
docker run -d --rm \
  --name econia_db_test \
  -e POSTGRES_PASSWORD=pgpw \
  -e POSTGRES_DB=econia \
  -p 37949:5432 postgres

export DATABASE_URL=postgres://postgres:pgpw@localhost:37949/econia

diesel database setup
```

Note that once the int types will be updated in the `schema.rs`, you'll probably also have to replace some `i64`s with `BigDecimal`s.

**Review by CRBl69 — CHANGES_REQUESTED** [review]
The changes look mainly good. Logic is good. Just a couple of very minor changes and then we're good to merge.

**Review by CRBl69 — APPROVED** [review]

**Pull request merged**